packages/composites/token-allocation/src/dataProvider.ts

```diff
@@ -2,11 +2,40 @@ import { Requester } from '@chainlink/ea-bootstrap'
 import { RequestConfig } from '@chainlink/types'
 import { ResponsePayload } from './types'
 
-export const getPriceProvider = (apiConfig: RequestConfig, jobRunID: string) => async (
-  symbols: string[],
-  quote: string,
-  withMarketCap = false,
-): Promise<ResponsePayload> => {
+const batchingSupport: { [name: string]: boolean } = {
+  COINGECKO: true,
+  COINMARKETCAP: true,
+  CRYPTOCOMPARE: true,
+  NOMICS: true,
+}
+
+export const getPriceProvider = (
+  jobRunID: string,
+  source: string,
+  apiConfig: RequestConfig,
+) => async (symbols: string[], quote: string, withMarketCap = false): Promise<ResponsePayload> => {
+  if (
+    batchingSupport[source.toUpperCase()] ||
+    (source.toUpperCase() === 'COINAPI' && quote.toUpperCase() === 'USD') // Special case for CoinAPI which only can batch USD quotes
```

Comment:
Since `COINAPI` isn't included in the variable where I am holding which EAs support batching (`batchingSupport`) this lets it pass through if the EA when  CoinAPI and the quote is USD, otherwise it doesn't batch.

Maybe it would be more clear to include `COINAPI` in batchingSupport and bail if quote isn't USD.